bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -74,6 +74,8 @@
     final double majorCompactionThreshold;
     final long majorCompactionInterval;
     long lastMajorCompactionTime;
+    final long forceCompactionInterval;
+    final AtomicLong lastMainCompactionTime;
```

Comment:
why are we calling it 'main' compaction? Don't we need to delay as long as we run *any* compaction?

Reply:
@jvrao - here main compaction will be Major compaction, if majorcompaction is
  disabled/suspended then it will be minorcompaction. If both major and minor are
  disabled/suspended then it will be just gc run.

I dont want to delay if major compaction is enabled and just minor compaction has run recently. Because if forcegc is triggered immediately without delay, then there is good chance of reclaiming enough space to become writable again.

Reply:
yes @jvrao That's right.
I will rename this as `lastCompactionTime` and add a comment there.